src/scripting/lua_attributes.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+	Copyright (C) 2009 - 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+/// Attribute registration system, mainly for objects with a lot of attributes, like units
+/// Not used for GUI2 widgets, as they're even more complicated with a deep hierarchy.
+
+#pragma once
+
+struct lua_State;
+class t_string;
+class vconfig;
+
+#include "config.hpp"
+#include "variable_info.hpp"
+#include "map/location.hpp"
```

Comment:
Alphabetical order - I know it's annoying to be reminded, but please get in the habit of doing it

Reply:
I do somewhat have a habit of doing it. The problem is that some of the files aren't already in alphabetical order, so I don't bother, and then I forget.